libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/routefetch/RoutesViewModel.kt

```diff
@@ -39,7 +54,23 @@ internal class RoutesViewModel : UIViewModel<Unit, RoutesAction>(Unit) {
             is RoutesAction.SetRoutes -> {
                 mapboxNavigation.setNavigationRoutes(action.routes, action.legIndex)
             }
+            is RoutesAction.StartNavigation -> {
+                if (mapboxNavigation.getNavigationRoutes().isEmpty()) {
+                    // fetching route if started from free drive
+                    fetchAndSetRoute(mapboxNavigation, state.destination) {
+                        startNavigation(mapboxNavigation)
+                    }
+                } else {
+                    startNavigation(mapboxNavigation)
+                }
+                return state.copy(navigationStarted = true)
+            }
```

Comment:
@kmadsen Any idea how can we better structure this?

Reply:
Yeah I was just staring at this too. I think this should be in a `NavigationStateViewModel`. It will have access to `MapboxNavigation` and routes. That view model will take over the `ReplayRoutesViewModel`, because sometimes we startReplayTripSessions

Reply:
OK, but how will `NavgationStateViewModel` fetch routes? 
When we start active navigation from Free Drive, routes might not be loaded for currently set Destination.
Should we add same method `fetchAndSetRoute` there as well?

Reply:
Wondering if we can resolve this with, making the destination be part of the `NavigationState` objects.

But I think it's fine and good to get this in and we can iterate on how to organize this logic